training/worker.py

```diff
@@ -13,21 +13,26 @@
 import service_pb2
 import service_pb2_grpc
 
+from threading import Lock
 
 from timemachine.lib import custom_ops, ops
 
 class Worker(service_pb2_grpc.WorkerServicer):
 
     def __init__(self):
-        self.state = None
+        self.states = {}
+        self.mutex = Lock()
 
     def ResetState(self, request, context):
-        self.state = None
+        self.mutex.acquire()
+        self.states = {}
```

Comment:
self.states.clear() 

Might be better here

Reply:
yeah i think it's a little clearer - not sure GCing the old object works when the ref exits, but probably have a single object is better.